From f347db08a1d199df3b2dfd45002bc64103d15fd2 Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Mon, 24 Apr 2017 11:38:03 +0300
Subject: [PATCH 0116/1239] arm64: dts: a8k: add sar device tree node

This patch adds the sar device tree nodes
with freq and boot_mode fields
to armada-7040-db and armada-8040-db.

Change-Id: I425736dc020f949662e342aafd37b99d51f52738
Signed-off-by: zachary zhang <zhangzg@marvell.com>
---
 arch/arm/dts/armada-7040-dev-info.dtsi | 73 ++++++++++++++++++++++++++
 arch/arm/dts/armada-70x0-db.dtsi       |  1 +
 arch/arm/dts/armada-8040-dev-info.dtsi | 64 ++++++++++++++++++++++
 arch/arm/dts/armada-80x0-db.dtsi       |  1 +
 4 files changed, 139 insertions(+)
 create mode 100644 arch/arm/dts/armada-7040-dev-info.dtsi
 create mode 100644 arch/arm/dts/armada-8040-dev-info.dtsi

diff --git a/arch/arm/dts/armada-7040-dev-info.dtsi b/arch/arm/dts/armada-7040-dev-info.dtsi
new file mode 100644
index 0000000000..42b63f0968
--- /dev/null
+++ b/arch/arm/dts/armada-7040-dev-info.dtsi
@@ -0,0 +1,73 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2017 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/ {
+	/* This should go only into devel boards */
+	compatible = "marvell,cp110";
+	sar {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		sar_fields {
+			compatible = "marvell,sample-at-reset";
+			reg = <0x4c 0x4e>;
+			chip_count = <2>;
+			bit_width = <5>;
+
+			freq {
+				key = "freq";
+				description = "CPU/DDR frequencies";
+				start-bit = <0>;
+				bit-length = <3>;
+				option-cnt = <7>;
+				options = "0x0", "1600/800 Mhz",
+					  "0x2", "1000/650 Mhz",
+					  "0x3", "1200/800 Mhz",
+					  "0x4", "1400/800 Mhz",
+					  "0x5", "600/800 Mhz",
+					  "0x6", "800/800 Mhz",
+					  "0x7", "1000/800 Mhz";
+				default = <0x3>;
+				status = "okay";
+			};
+			boot_mode {
+				key = "boot_mode";
+				description = "Boot mode options";
+				start-bit = <4>;
+				bit-length = <6>;
+				option-cnt = <14>;
+				options = "0xE", "NAND IHB BW-8bit, PS-4KB, ECC-4bit",
+					  "0xF", "NAND IHB BW-8bit, PS-4KB, ECC-8bit",
+					  "0x28", "AP_SD",
+					  "0x29", "CP_SD",
+					  "0x2A", "AP_EMMC",
+					  "0x2B", "CP_EMMC",
+					  "0x30", "AP_SPI 24bits",
+					  "0x31", "AP SPI BR-Disabled 24bits",
+					  "0x32", "CP_SPI_1 24bits",
+					  "0x33", "CP_SPI_1 32bits",
+					  "0x34", "CP_SPI_0 24bits",
+					  "0x35", "CP_SPI_0 32bits",
+					  "0x36", "CP_SPI_1 BR-Disabled 24bits",
+					  "0x37", "CP_SPI_1 BR-Disabled 32bits";
+				default = <0x32>;
+				status = "okay";
+			};
+		};
+	};
+};
diff --git a/arch/arm/dts/armada-70x0-db.dtsi b/arch/arm/dts/armada-70x0-db.dtsi
index dce8ff1423..2f6b0dac1b 100644
--- a/arch/arm/dts/armada-70x0-db.dtsi
+++ b/arch/arm/dts/armada-70x0-db.dtsi
@@ -4,6 +4,7 @@
  */
 
 #include "armada-70x0.dtsi" /* include SoC device tree */
+#include "armada-7040-dev-info.dtsi"
 
 / {
 	model = "DB-ARMADA-70x0";
diff --git a/arch/arm/dts/armada-8040-dev-info.dtsi b/arch/arm/dts/armada-8040-dev-info.dtsi
new file mode 100644
index 0000000000..a8a51f225a
--- /dev/null
+++ b/arch/arm/dts/armada-8040-dev-info.dtsi
@@ -0,0 +1,64 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2017 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/ {
+	/* This should go only into devel boards */
+	compatible = "marvell,cp110";
+	sar {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		sar_fields {
+
+			compatible = "marvell,sample-at-reset";
+			reg = <0x4c 0x4e>;
+			chip_count = <2>;
+			bit_width = <5>;
+
+			freq {
+				key = "freq";
+				description = "CPU/DDR and PIDI frequencies";
+				start-bit = <0>;
+				bit-length = <4>;
+				option-cnt = <4>;
+				options = "0x1", "CPU/DDR = 0x1:  2000/1050 Mhz, PIDI = 0: 1Ghz",
+					  "0x3", "CPU/DDR = 0x3:  1800/1050 Mhz, PIDI = 0: 1Ghz",
+					  "0x4", "CPU/DDR = 0x4:  1600/1050 Mhz, PIDI = 0: 1Ghz",
+					  "0xe", "CPU/DDR = 0x34: 1300/800 Mhz,  PIDI = 1: 500Mhz";
+				default = <0x1>;
+				status = "okay";
+			};
+			boot_mode {
+				key = "boot_mode";
+				description = "Boot mode options";
+				start-bit = <4>;
+				bit-length = <6>;
+				option-cnt = <7>;
+				options = "0xE", "CP1_NAND IHB BW-8bit, PS-4KB, ECC-4bit\t(supported configuration: D)",
+					  "0xF", "CP1_NAND IHB BW-8bit, PS-4KB, ECC-8bit\t(supported configuration: D)",
+					  "0x29", "CP0_SD\t\t\t\t\t(supported configurations: C, D and E)",
+					  "0x2A", "AP_EMMC\t\t\t\t\t(supported configurations: A, B, C and D)",
+					  "0x2B", "CP0_EMMC\t\t\t\t\t(supported configurations: A and B)",
+					  "0x30", "AP_SPI 24bits\t\t\t\t(supported configurations: E and A0)",
+					  "0x32", "CP1_SPI_1 24bits\t\t\t\t(supported configurations: A, B, C and E)";
+				default = <0x32>;
+				status = "okay";
+			};
+		};
+	};
+};
diff --git a/arch/arm/dts/armada-80x0-db.dtsi b/arch/arm/dts/armada-80x0-db.dtsi
index bd505b1a0b..a40c3edba0 100644
--- a/arch/arm/dts/armada-80x0-db.dtsi
+++ b/arch/arm/dts/armada-80x0-db.dtsi
@@ -4,6 +4,7 @@
  */
 
 #include "armada-8040.dtsi" /* include SoC device tree */
+#include "armada-8040-dev-info.dtsi"
 
 / {
 	compatible = "marvell,armada-80x0-db", "marvell,armada-8040";
-- 
2.29.0

